### PR TITLE
 [Enhancement] Update routine load parameters

### PR DESCRIPTION
**Pull request by zaorangyang**
This PR removes the routine_load_thread_pool_size parameter and the number of routine load tasks will be completely controlled by max_routine_load_task_num_per_be.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [x] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 15 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadMgr.java|12|15|80.00%|[247, 255, 256]|

File: docs/administration/Configuration.md [hunk 1]
```diff
@@ -109,7 +109,6 @@ ADMIN SET FRONTEND CONFIG ("key" = "value");
 | label_keep_max_second                   | s    | 259200                                          | The maximum duration the labels of load jobs that have been completed and are in the FINISHED or CANCELLED state can be retained in the StarRocks system. The default value is 3 days. After this duration expires, the labels will be deleted. This parameter applies to all types of load jobs. Unit: seconds. A value too large consumes a lot of memory. |
 | max_routine_load_job_num                | -    | 100                                             | The maximum number of Routine Load jobs in a StarRocks cluster. |
 | max_routine_load_task_concurrent_num    | -    | 5                                               | The maximum number of concurrent tasks for each Routine Load job. |
-| max_routine_load_task_num_per_be        | -    | 5                                               | The maximum number of concurrent Routine Load tasks that can run for each BE. The value must be less than or equal to the BE configuration item `routine_load_thread_pool_size`. |
```
Comment (meegoo) [inline docs/administration/Configuration.md:112]: why remove it?

File: be/src/runtime/routine_load/routine_load_task_executor.h [hunk 2]
```diff
@@ -59,22 +59,27 @@ class RoutineLoadTaskExecutor {
...
-              _data_consumer_pool(10) {
+    RoutineLoadTaskExecutor(ExecEnv* exec_env) : _exec_env(exec_env), _data_consumer_pool(10) {
         REGISTER_GAUGE_STARROCKS_METRIC(routine_load_task_count, [this]() {
             std::lock_guard<std::mutex> l(_lock);
             return _task_map.size();
         });
-
+        auto st = ThreadPoolBuilder("routine_load")
+                          .set_min_threads(1)
+                          .set_max_threads(500)
```
Comment (meegoo) [inline be/src/runtime/routine_load/routine_load_task_executor.h:69]: no need to limit

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 15 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadMgr.java|12|15|80.00%|[247, 255, 256]|

File: docs/administration/Configuration.md (hunk 1 above)
Comment (zaorangyang) [reply docs/administration/Configuration.md]: > 

A mistake. Updated.

File: be/src/runtime/routine_load/routine_load_task_executor.h (hunk 2 above)
Comment (zaorangyang) [reply be/src/runtime/routine_load/routine_load_task_executor.h]: > 

updated

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 15 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadMgr.java|12|15|80.00%|[247, 255, 256]|

File: be/src/runtime/routine_load/routine_load_task_executor.h [hunk 3]
```diff
@@ -59,22 +59,23 @@ class RoutineLoadTaskExecutor {
...
-              _thread_pool("routine_load", config::routine_load_thread_pool_size,
-                           config::routine_load_thread_pool_size),
-              _data_consumer_pool(10) {
+    RoutineLoadTaskExecutor(ExecEnv* exec_env) : _exec_env(exec_env), _data_consumer_pool(10) {
         REGISTER_GAUGE_STARROCKS_METRIC(routine_load_task_count, [this]() {
             std::lock_guard<std::mutex> l(_lock);
             return _task_map.size();
         });
-
+        auto st = ThreadPoolBuilder("routine_load").build(&_thread_pool);
```
[the remark this answers is not in the document]
Comment (zaorangyang) [reply be/src/runtime/routine_load/routine_load_task_executor.h:67]: @chaoyli 
image

[the remark this answers is not in the document]
Comment (meegoo) [reply be/src/runtime/routine_load/routine_load_task_executor.h:67]: max thread should set to unlimited, only control by FE

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 15 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/routineload/RoutineLoadMgr.java|12|15|80.00%|[247, 255, 256]|

**Review by meegoo — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by zaorangyang** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**